hydra-plutus/src/Hydra/Contract/Deposit.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -fno-specialize #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:conservative-optimisation #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+-- Plutus core version to compile to. In babbage era, that is Cardano protocol
+-- version 7 and 8, only plutus-core version 1.0.0 is available.
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:target-version=1.0.0 #-}
+
+-- | The validator used to deposit and recover locked funds
+module Hydra.Contract.Deposit where
+
+import PlutusTx.Prelude
+
+import Hydra.Cardano.Api (PlutusScriptVersion (PlutusScriptV2))
+import Hydra.Contract.Commit (Commit)
+import Hydra.Contract.DepositError (
+  DepositError (
+    DepositDeadlineNotReached,
+    DepositNoLowerBoundDefined,
+    IncorrectDepositHash
+  ),
+ )
+import Hydra.Contract.Error (errorCode)
+import Hydra.Contract.Head (hashPreSerializedCommits, hashTxOuts)
+import Hydra.Plutus.Extras (ValidatorType, scriptValidatorHash, wrapValidator)
+import PlutusLedgerApi.V2 (
+  CurrencySymbol,
+  Datum (Datum),
+  Extended (Finite),
+  Interval (ivFrom),
+  LowerBound (LowerBound),
+  POSIXTime,
+  Redeemer (Redeemer),
+  ScriptContext (..),
+  ScriptHash,
+  SerialisedScript,
+  serialiseCompiledCode,
+  txInfoOutputs,
+  txInfoValidRange,
+ )
+import PlutusTx (CompiledCode, toBuiltinData)
+import PlutusTx qualified
+
+data DepositRedeemer
+  = -- | Claims already deposited funds.
+    Claim
+  | -- | Recovers m number of deposited outputs.
+    Recover Integer
+
+PlutusTx.unstableMakeIsData ''DepositRedeemer
+
+-- | Deposit datum containing HeadId, deadline and a list of deposits.
+type DepositDatum = (CurrencySymbol, POSIXTime, [Commit])
+
+-- | v_deposit validator checks
+--
+-- * Claim redeemer -> no checks are needed
```

Comment:
This is not true. There will be checks, but it's not implemented.

Must: update `Claim -> False` below to not allow for free spending via `Claim`

Reply:
Do you mean checks that outputs are locked correctly? Shall I add them in this pr?

Reply:
No need to add checks right. Making it always fail with `False` is good enough for this PR.